scanpy/external/_tools/_annotator.py

```diff
@@ -0,0 +1,116 @@
+    return_nonzero_annotations : bool, optional (default=True)
+        If true return scores only for cell types that have no zero scores.
+    p_threshold : float
+        A threshold for accepting the annotations. Annotations that have FDR
+        value bellow this threshold are used.
+    p_value_fun : str, optional (defaults: binom)
+        A function that calculates a p-value. It can be either
+        `binom` that uses binom.sf or
+        `hypergeom` that uses hypergeom.sf.
+    z_threshold: float
+        The threshold for selecting the gene from gene expression data.
+        For each cell the attributes with z-value above this value are selected.
+    scoring : str, optional (default = "exp_ratio")
+        Scoring method for cell type scores. Available scores are:
+        - exp_ratio - the proportion of genes typical for a cell type expressed
+        in the cell
+        - sum_of_expressed_markers - the sum of expressions of genes typical for
+        a cell type
+        - log_fdr - negative of the logarithm of an false discovery rate (FDR)
+        value
+        - log_p_value - negative of the logarithm of an p-value
+    normalize : bool, optional (default = False)
+        If this parameter is True data will be normalized during the
+        a process with a log CPM normalization.
+       That method works correctly data needs to be normalized.
+        Set this `normalize` on True if your data are not normalized already.
+
+    Returns
+    -------
+    pd.DataFrame
+        Cell type for each cell for each cell. The result is a sore matrix that
+        tells how probable is each cell type for each cell. Columns are cell
+        types and rows are cells.
+
+    Example
+    -------
+
+    Here is the example of annotation of dendritic cells based on their gene
+    expressions. For annotation, we use data by Villani et al. (2017) and marker
+    genes by Franzén et al. (2019).
+
+    >>> import pandas as pd
+    >>> from scanpy import AnnData
+    >>> from scanpy.external.tl import annotator
+    >>> import urllib.request
+    >>>
+    >>> # download data in a temporary directory
+    >>> file_name_data, _ = urllib.request.urlretrieve(
+    ...     "https://github.com/PrimozGodec/cell-annotation/releases/download/"
+    ...     "0.1.0/DC_expMatrix_DCnMono.csv.gz")
+    >>> file_name_markers, _ = urllib.request.urlretrieve(
+    ...     "https://github.com/PrimozGodec/cell-annotation/releases/download/"
+    ...     "0.1.0/panglao_gene_markers.csv.gz")
+    >>>
+    >>> # read data with pandas
+    >>> df = pd.read_csv(file_name_data, compression="gzip").iloc[:, :-2]
+    >>> df_markers = pd.read_csv(file_name_markers, compression="gzip")
+    >>> # current markers have wrong name for Gene column
+    >>> df_markers = df_markers.rename(columns={'Name': 'Gene'})
+    >>> # we need just marker genes for human
+    >>> df_markers = df_markers[df_markers["Organism"] == "Human"]
```

Comment:
Prepare the data such that the markers are already human only.

Reply:
Done